pupil_src/shared_modules/calibration_routines/gaze_mappers.py

```diff
@@ -76,68 +76,128 @@ class Binocular_Gaze_Mapper_Base(Gaze_Mapping_Plugin):
 
     def __init__(self, g_pool):
         super().__init__(g_pool)
+        self._cfg = {'min_pupil_confidence': 0.6,
+                     'temporal_cutoff': 0.03,
+                     'timestamp_reset_threshold': 5.0,
+                     'process_low_confidence': True,
+                     'sample_cutoff': 10}
+        self._state = {'cache': deque(),
+                       'cache_ts': deque(),
+                       'cache_id': deque(),
+                       'cache_conf': deque()}
```

Comment:
I would recommend to extract the state into its own class. Additionally, instead of managing 4 separate univariate deques, I would recommend to use a single queue with multi-variate values (if at all necessary). The only real reason to have more than just the datum's deque is `in_ix = np.searchsorted(self._state["cache_ts"], p["timestamp"])`, and even there one could argue that one could replace the extra deque with a list comprehension, building the list on demand.

The main argument here is that keeping 4 lists coherent is much more work (in the sense of maintenance) than managing a single list.

Reply:
Agreed. They were left over from when I was pulling my hair out trying to figure out why I was generating garbage data. The state is simplified to a single cache of data samples.